iterative/utils/analytics.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shirou/gopsutil/host"
+	"github.com/sirupsen/logrus"
+)
+
+func SystemInfo() map[string]interface{} {
+	hostStat, _ := host.Info()
+	return map[string]interface{}{
+		"os":               hostStat.OS,
+		"platform":         hostStat.Platform,
+		"platform_version": hostStat.PlatformVersion,
+	}
+}
+
+func TaskDuration(logs string) float64 {
+	regex := regexp.MustCompile(`\w{3} \d{2} \d{2}:\d{2}:\d{2}`)
+	matches := regex.FindAllString(logs, -1)
+	taskDuration := 0.0
+
+	if len(matches) > 1 {
+		layout := "Mar 07 06:43:27"
+		t1, _ := time.Parse(layout, matches[len(matches)-1])
+		t0, _ := time.Parse(layout, matches[0])
+		taskDuration = t1.Sub(t0).Seconds()
```

Comment:
@0x2b3bfa0 currently task has a date format that is not ideal to parse it.

Reply:
If you want this feature, you may prefer to write an object to object storage with the start date and the end date.

Reply:
Yes, I find total time average a very useful insight

Reply:
Low–hanging fruit: if you want to know whether people uses spot instance recovery, you can always `if spot: len([]logs)` because every instance in the chain creates a log.